packages/nuxt/src/app/components/nuxt-link.ts

```diff
@@ -120,6 +120,79 @@ export function defineNuxtLink (options: NuxtLinkOptions) {
     return resolvedPath
   }
 
+  function useNuxtLink (props: NuxtLinkProps) {
```

Comment:
Just FYI this logic has some issues, see https://github.com/nuxt/nuxt/issues/25532

I wouldn't consider fixing them as part of this PR but it will likely be updated in the future.

Reply:
Exactly. ❤️  Just wanted to  check that this implementation would be compatible with the tree-shaking/bundle size improvements you have in mind.